### PR TITLE
optionally save full simulation output

### PR DESCRIPTION
**Pull request by haakon-e**
## Changes

With this PR, you can add the following entries to the `output` config. The first entry is a flag to specify whether to save forward model data or not. If that is set to true, you can optionally specify only a subset of iterations to save.
```julia
config["save_tc_output"] = true  # save sims
config["save_tc_iterations"] = [1, 4]  # save subset of iterations
```

Data from both training and validation will be saved to separate directories.

The output data is stored in the following directory structure:
```
<results_folder>/timeseries.<suffix>/iter_<iteration>/Output.<case>.<case_id>_<ens_i>
```
where `suffix` is either `train` or `validation`, and `case_id` is an integer that ensures that if the same case in run multiple times the output is stored in unique directories.

## Checklist before requesting a review / merging
- [x] I have formatted the code using `julia --project=.dev .dev/climaformat.jl .`
- [x] I have updated all test manifests using `julia --project .dev/up_deps.jl .` with Julia 1.7.3.
- [x] If core features are added, I have added appropriate test coverage.
- [x] If new functions and structs are added, they are documented through docstrings.

### CODE REVIEW

**Comment by haakon-e** [comment]
before merge: need to check behavior in the case when there are multiple configurations with the same name.

File: src/Pipeline.jl [hunk 1]
```diff
@@ -453,6 +453,24 @@ function ek_update(
...
+                dst_root = joinpath(outdir_path, "timeseries/$case/iter_$iteration")
+                mkpath(dst_root)
+                if isdir(src)
+                    mv(src, joinpath(dst_root, "Output.$case.$i"))
+                else
+                    @warn("sim dir not found: $src. Cannot save forward model timeseries.")
+                end
+            end
+        end
+    end
```
Comment (ilopezgp) [inline src/Pipeline.jl:472]: can you encapsulate this in an external function? Also, we currently have in TurbulenceConvectionUtils.jl the function `save_full_ensemble_data`. Can you use or modify that instead? We have no coverage or use of the function at the moment.

File: src/Pipeline.jl [hunk 2]
```diff
@@ -725,6 +743,25 @@ function versioned_model_eval(version::Union{String, Int}, outdir_path::String,
...
+            case = cases[case_ind][1]
+            savedir = joinpath(outdir_path, "timeseries/$case")
+            mkpath(savedir)
+            if isdir(sim_dir)
+                cp(sim_dir, joinpath(savedir, "Output.$case.$version"))
+            else
+                @warn("sim directory not found: $sim_dir")
+            end
+        end
+    end
```
Comment (ilopezgp) [inline src/Pipeline.jl:763]: same here.

File: src/Pipeline.jl (hunk 1 above)
Comment (haakon-e) [reply src/Pipeline.jl:472]: Thanks for the suggestion, I forgot we still had `save_full_ensemble_data` lying around. That function is outdated, so I'll move this new code into a function a delete the old one.

**Comment by ilopezgp** [comment]
You need to rebase with main so the tests pass, I think

File: src/TurbulenceConvectionUtils.jl [hunk 3]
```diff
@@ -521,29 +522,39 @@ function get_gcm_les_uuid(
...
-            cp(tmp_data_path, save_data_path)
-            # namefile
-            tmp_namefile_path = namelist_directory(sim_dir, scm_name)
-            save_namefile_path = namelist_directory(ens_i_path, scm_name)
-            cp(tmp_namefile_path, save_namefile_path)
-        end
+""" 
+    save_tc_data(cases, outdir_path, sim_dirs, version, suffix)
+
+Save full output timeseries in `<results_folder>/timeseries.<suffix>/iter_<iteration>/Output.<case>.<case_id>_<ens_i>`.
```
Comment (ilopezgp) [inline src/TurbulenceConvectionUtils.jl:528]:
```suggestion
Save full TC.jl output in `<results_folder>/timeseries.<suffix>/iter_<iteration>/Output.<case>.<case_id>_<ens_i>`.
```

File: src/TurbulenceConvectionUtils.jl [hunk 4]
```diff
@@ -521,29 +522,39 @@ function get_gcm_les_uuid(
...
+
+A vector of iteration indices can be specified in `config["output"]["save_tc_iterations"]` to save only a subset of the data.
```
Comment (ilopezgp) [inline src/TurbulenceConvectionUtils.jl:530]: It is better to explain what the inputs are and how they are used, instead.

File: src/TurbulenceConvectionUtils.jl [hunk 5]
```diff
@@ -521,29 +522,39 @@ function get_gcm_les_uuid(
...
+"""
+function save_tc_data(config, outdir_path, sim_dirs, version, suffix)
+    cases = config["reference"]["case_name"]
+    @assert length(cases) == length(sim_dirs) "this should be true"
```
Comment (ilopezgp) [inline src/TurbulenceConvectionUtils.jl:534]:
```suggestion
    @assert length(cases) == length(sim_dirs) "The number of cases $(length(cases)) should equal the number of TC output directories $(length(sim_dirs))."
```

File: experiments/scm_pycles_pipeline/config.jl [hunk 6]
```diff
@@ -41,7 +41,8 @@ end
 function get_output_config()
     config = Dict()
     config["outdir_root"] = pwd()
-    config["overwrite_scm_file"] = false # Flag for overwritting SCM input file
+    config["overwrite_scm_file"] = false  # Flag for overwriting SCM input file
+    config["save_forward_sim_ts"] = false  # flag for saving full data from forward model evaluations.
```
Comment (ilopezgp) [inline experiments/scm_pycles_pipeline/config.jl:45]: Shouldn't this be `"save_tc_output"`?

**Review by ilopezgp — APPROVED** [review]
LGTM

**Pull request merged**